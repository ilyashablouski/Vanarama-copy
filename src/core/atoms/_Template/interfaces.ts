import { IBaseProps } from '../../interfaces/base';

export interface ITemplateProps extends IBaseProps {}
