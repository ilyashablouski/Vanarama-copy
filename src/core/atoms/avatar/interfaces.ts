import { IBaseProps } from '../../interfaces/base';
import { TSize } from '../../../types/size';

export interface IAvatarProps extends IBaseProps {
  src?: string;
  size?: TSize;
  altText?: string;
}
