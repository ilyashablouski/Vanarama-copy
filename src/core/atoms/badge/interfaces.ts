import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';
import { TSize } from '../../../types/size';

export interface IBadgeProps extends IBaseProps {
  size?: TSize;
  color?: TColor;
  label?: string;
}
