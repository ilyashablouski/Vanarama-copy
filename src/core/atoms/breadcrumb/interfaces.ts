import { IBaseProps } from '../../interfaces/base';
import { ILink } from '../../interfaces/link';

export interface IBreadcrumbProps extends IBaseProps {
  items?: ILink[];
}
