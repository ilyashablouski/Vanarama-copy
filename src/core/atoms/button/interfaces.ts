import { ButtonHTMLAttributes } from 'react';
import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';
import { TSize } from '../../../types/size';

export interface IButtonProps
  extends IBaseProps,
    ButtonHTMLAttributes<HTMLButtonElement> {
  color?: TColor;
  fill?: 'solid' | 'outline' | 'clear';
  icon?: React.ReactNode;
  iconColor?: TColor;
  iconPosition?: 'before' | 'after';
  label?: string | React.ReactNode;
  round?: boolean;
  size?: TSize;
  withoutDefaultClass?: boolean;
  customCTAColor?: string;
}
