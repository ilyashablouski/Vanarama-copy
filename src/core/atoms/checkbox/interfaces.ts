import { InputHTMLAttributes, ReactElement } from 'react';
import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';

export interface ICheckboxProps
  extends IBaseProps,
    InputHTMLAttributes<HTMLInputElement> {
  /**
   * A color for the checkbox.
   */
  color?: TColor;
  /**
   * A unique ID for the checkbox node - this is required here because the
   * underlying checkbox is hidden and therefore you can only click on the label
   * and the label needs to know which input it corresponds to.
   */
  id: string;
  /**
   * 	Invalid state of the checkbox.
   */
  invalid?: boolean;
  /**
   * A name for the input node.
   */
  label: string | (string | ReactElement)[];
  /**
   * Outlined state of the checkbox.
   */
  outline?: boolean;
  /**
   * A ref to attach to the checkbox.
   */
  ref?: React.Ref<HTMLInputElement>;
}
