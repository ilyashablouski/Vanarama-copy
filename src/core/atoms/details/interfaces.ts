import { IBaseProps } from '../../interfaces/base';

export interface IDetailsProps extends IBaseProps {
  summary?: string;
  content?: string;
  open?: boolean;
}
