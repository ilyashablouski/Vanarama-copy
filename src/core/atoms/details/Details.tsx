import React, { FC, memo } from 'react';
import cx from 'classnames';

import { IDetailsProps } from './interfaces';

const Details: FC<IDetailsProps> = memo(props => {
  const { className, children, summary, open } = props;

  return (
    <details className={cx('details', className)} open={open}>
      <summary className="details--summary">{summary}</summary>
      <div className="details--content">{children}</div>
    </details>
  );
});

Details.displayName = 'Details';

export default Details;
