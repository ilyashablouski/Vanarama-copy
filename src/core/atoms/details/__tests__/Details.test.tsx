import React from 'react';
import renderer from 'react-test-renderer';

import Text from '../../text';

import Details from '..';
import { IDetailsProps } from '../interfaces';

const optionalProps = {
  summary: 'Summary',
  open: false,
};

const children = <Text>Text text text text text text text text.</Text>;

function getComponent(props?: IDetailsProps) {
  return renderer.create(<Details {...props}>{children}</Details>).toJSON();
}

describe('<Details />', () => {
  it('renders correctly', () => {
    const tree = getComponent();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with optional props', () => {
    const tree = getComponent(optionalProps);
    expect(tree).toMatchSnapshot();
  });
});
