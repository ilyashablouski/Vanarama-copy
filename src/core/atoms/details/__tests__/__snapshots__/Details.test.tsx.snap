// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> renders correctly 1`] = `
<details
  className="details"
>
  <summary
    className="details--summary"
  />
  <div
    className="details--content"
  >
    <span
      className="text -regular -darker"
    >
      Text text text text text text text text.
    </span>
  </div>
</details>
`;

exports[`<Details /> renders correctly with optional props 1`] = `
<details
  className="details"
  open={false}
>
  <summary
    className="details--summary"
  >
    Summary
  </summary>
  <div
    className="details--content"
  >
    <span
      className="text -regular -darker"
    >
      Text text text text text text text text.
    </span>
  </div>
</details>
`;
